pkg/controller/volume/attach_detach_controller_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
+	"k8s.io/kubernetes/pkg/client/testing/core"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+func Test_Run_Positive(t *testing.T) {
+	// Arrange
+	fakeKubeClient := createTestClient()
+	adc, err := NewAttachDetachController(
+		fakeKubeClient,
+		nil,           /* podInformer */
+		nil,           /* nodeInformer */
+		nil,           /* pvcInformer */
+		nil,           /* pvInformer */
+		5*time.Minute, /* resyncPeriod */
+		nil,           /* cloud */
+		nil /* plugins */)
+
+	// Act
+	go adc.Run(wait.NeverStop)
```

Comment:
Dumb sanity check on NewAttachDetachController. The goroutine for Run was doing nothing, removed it.